### PR TITLE
get_ecdsa_sig_rs_bytes(): free value of d2i_ECDSA_SIG() before return

### PR DESCRIPTION
**Pull request by PeiweiHu**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Review by t8m — APPROVED** [review]

File: test/acvp_test.c [hunk 1]
```diff
@@ -217,8 +217,10 @@ static int get_ecdsa_sig_rs_bytes(const unsigned char *sig, size_t sig_len,
         return 0;
     r1 = ECDSA_SIG_get0_r(sign);
     s1 = ECDSA_SIG_get0_s(sign);
-    if (r1 == NULL || s1 == NULL)
+    if (r1 == NULL || s1 == NULL) {
+        ECDSA_SIG_free(sign);
         return 0;
```
Comment (mattcaswell) [inline test/acvp_test.c:222]: Why not just `goto err`?

File: test/acvp_test.c [hunk 2]
```diff
@@ -559,8 +561,10 @@ static int get_dsa_sig_rs_bytes(const unsigned char *sig, size_t sig_len,
     if (sign == NULL)
         return 0;
     DSA_SIG_get0(sign, &r1, &s1);
-    if (r1 == NULL || s1 == NULL)
+    if (r1 == NULL || s1 == NULL) {
+        DSA_SIG_free(sign);
         return 0;
```
Comment (mattcaswell) [inline test/acvp_test.c:566]: `goto err`

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Review by paulidale — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by davidben** [comment]
Alternatively, those checks can just be deleted. No successful `d2i_ECDSA_SIG` or `d2i_DSA_SIG` call can produce an object with missing `r` or `s` values. The parameters aren't optional and the function always fills them in:
https://github.com/openssl/openssl/blob/master/crypto/ec/ec_asn1.c#L1226

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**